proseco/guide.py

```diff
@@ -340,6 +349,45 @@ def get_initial_guide_candidates(self):
         return cand_guides
 
 
+
+def check_fid_trap(cand_stars, fids, dither):
+    """
+    Search for guide stars that would cause the fid trap issue and mark as spoilers.
+    """
+
+    spoilers = np.zeros(len(cand_stars)).astype(bool)
+    rej = []
+
+    if len(fids) == 0:
+        return spoilers, []
+
+    if not isinstance(dither, ACABox):
+        dither = ACABox(dither)
+    bad_row = GUIDE_CHAR.fid_trap['row']
+    bad_col= GUIDE_CHAR.fid_trap['col']
+    fid_margin = GUIDE_CHAR.fid_trap['margin']
+
+    # Check to see if the fid is in the zone that's a problem for the trap and if there's
+    # a star that can cause the effect in the readout regiser
+    for fid in fids:
+        incol = abs(fid['col'] - bad_col) < fid_margin
+        inrow = (fid['row'] < 0) & (fid['row'] > bad_row)
+        if (incol & inrow):
```

Comment:
Should be `if incol and inrow:`  (not a bitwise `&` but rather a logical `and`).  And no extraneous parens for simple expressions like this.